graphql/handler/apollofederatedtracingv1/logger/logger.go

```diff
@@ -0,0 +1,31 @@
+package tracing_logger
+
+import "log"
+
+// Logger is an interface that can be implemented to log errors that occur during the tracing process
+// This can use the default Go logger or a custom logger (e.g. logrus or zap)
+type Logger interface {
+	Print(args any)
+	Println(args any)
+	Printf(format string, args any)
+}
+
+func New() *NoopLogger {
```

Comment:
```suggestion
// This will create a logger that discards all messages (i.e. does not output)
func NewNoopLogger() *NoopLogger {
```
I would rename this, otherwise folks who read the package name + `New()` will think they are going to get some output.